ExampleMod/Content/Projectiles/Minions/ExampleSimpleMinion.cs

```diff
@@ -70,13 +70,12 @@ public class ExampleSimpleMinionItem : ModItem
 			Item.shoot = ModContent.ProjectileType<ExampleSimpleMinion>(); // This item creates the minion projectile.
 		}
 
-		public override bool Shoot(Player player, ref Vector2 position, ref float speedX, ref float speedY, ref int type, ref int damage, ref float knockBack) {
+		public override void ModifyShootStats(Player player, ref Vector2 position, ref Vector2 velocity, ref int type, ref int damage, ref float knockBack) {
 			// This is needed so the buff that keeps your minion alive and allows you to despawn it properly applies
 			player.AddBuff(Item.buffType, 2);
 
```

Comment:
Shouldn't `player.AddBuff(Item.buffType, 2);` be in the separate Shoot hook? Since it has nothing to do with modifying.

Reply:
Arguably you could say that it shouldn't `Shoot` either, since it has nothing to do with shooting. Vanilla doesn't even add minion buffs in the shooting code, and the closest hook to the place were vanilla applies its minion buffs would be `CanUseItem`, which works because the code that applies the buff doesn't actually care if the item can be used or not. Not sure if that would be proper hook usage though.